src/source-nfq.c

```diff
@@ -915,13 +911,12 @@ int NFQParseAndRegisterQueues(const char *queues)
  *  \param number idx of the queue in our array
  *
  *  \retval ptr pointer to the NFQThreadVars at index
- *  \retval NULL on error
  */
 void *NFQGetQueue(int number)
 {
-    if (unlikely(number < 0 || number >= receive_queue_num || g_nfq_q == NULL))
-        return NULL;
-
+    assert(number < 0);
```

Comment:
It can remain one line with `||`
And it looks like you are asserting the negation of what should be asserted, right ?

And I think we would like to use our macro `BUG_ON` rather than `assert`

Reply:
Something like
```
BUG_ON(number > 0 && number < receive_queue_num && g_nfq_q != NULL)
```

Reply:
Sorry, I really forgot to invert the conditions, but think there should be a BUG_ON for each condition that defines exactly what happened

Reply:
```
        #define BUG_ON(x) do {      \
            if (((x))) {            \
                fprintf(stderr, "BUG at %s:%d(%s)\n", __FILE__, __LINE__, __func__);    \
                fprintf(stderr, "Code: '%s'\n", xstr((x)));                             \
                exit(EXIT_FAILURE); \
            }                       \
        } while(0)
     ```
     I may be wrong, but it seems there is no need to invert the condition
     We want detect if (number < 0 || number >= receive_queue_num) and if g_nfq_q is NULL

Reply:
You're correct --- `BUG_ON` uses the condition `x` to trigger the issue.

Whether a single `BUG_ON` or not is used depends on context. In this case, I'd recommend a single `BUG_ON` statement.